kubelet/datadog_checks/kubelet/kubelet.py

```diff
@@ -429,7 +427,18 @@ def _retrieve_stats(self):
             return {}
 
     def _report_node_metrics(self, instance_tags):
-        node_spec = self._retrieve_node_spec()
+        try:
+            node_resp = self._retrieve_node_spec()
+            node_resp.raise_for_status()
+        except requests.HTTPError as e:
+            if node_resp.status_code == 404:
+                # ignore HTTPError, for supporting k8s >= 1.18 in a degrated mode
+                # in 1.18 the /spec can be reactivated from the kubelet config
+                # in 1.19 the /spec will removed.
+                self.log.warning('kubelet check "/spec" endpoint not available')
```

Comment:
I don't think it's super useful to log this in a loop.

Reply:
@therve PR updated according to your comment